tests/components/zwave_js/test_climate.py

```diff
@@ -0,0 +1,276 @@
+                "prevValue": 1,
+            },
+        },
+    )
+    node.receive_event(event)
+
+    state = hass.states.get(CLIMATE_RADIO_THERMOSTAT_ENTITY)
+    assert state.state == HVAC_MODE_COOL
+    assert state.attributes[ATTR_TEMPERATURE] == 22.8
+
+    # Test heat_cool mode update from value updated event
+    event = Event(
+        type="value updated",
+        data={
+            "source": "node",
+            "event": "value updated",
+            "nodeId": 13,
+            "args": {
+                "commandClassName": "Thermostat Mode",
+                "commandClass": 64,
+                "endpoint": 1,
+                "property": "mode",
+                "propertyName": "mode",
+                "newValue": 3,
+                "prevValue": 1,
+            },
+        },
+    )
+    node.receive_event(event)
+
+    state = hass.states.get(CLIMATE_RADIO_THERMOSTAT_ENTITY)
+    assert state.state == HVAC_MODE_HEAT_COOL
+    assert state.attributes[ATTR_TARGET_TEMP_HIGH] == 22.8
+    assert state.attributes[ATTR_TARGET_TEMP_LOW] == 22.2
+
+    client.async_send_command.reset_mock()
+
+    # Test setting temperature with heat_cool
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_TEMPERATURE,
+        {
+            ATTR_ENTITY_ID: CLIMATE_RADIO_THERMOSTAT_ENTITY,
+            ATTR_TARGET_TEMP_HIGH: 30,
+            ATTR_TARGET_TEMP_LOW: 25,
+        },
+        blocking=True,
+    )
+
+    assert len(client.async_send_command.call_args_list) == 2
+    args = client.async_send_command.call_args_list[0][0][0]
+    assert args["command"] == "node.set_value"
+    assert args["nodeId"] == 13
+    assert args["valueId"] == {
+        "commandClassName": "Thermostat Setpoint",
+        "commandClass": 67,
+        "endpoint": 1,
+        "property": "setpoint",
+        "propertyKey": 1,
+        "propertyName": "setpoint",
+        "propertyKeyName": "Heating",
+        "metadata": {
+            "type": "number",
+            "readable": True,
+            "writeable": True,
+            "unit": "°F",
+            "ccSpecific": {"setpointType": 1},
+        },
+        "value": 72,
+    }
+    assert args["value"] == 77
+
+    args = client.async_send_command.call_args_list[1][0][0]
+    assert args["command"] == "node.set_value"
+    assert args["nodeId"] == 13
+    assert args["valueId"] == {
+        "commandClassName": "Thermostat Setpoint",
+        "commandClass": 67,
+        "endpoint": 1,
+        "property": "setpoint",
+        "propertyKey": 2,
+        "propertyName": "setpoint",
+        "propertyKeyName": "Cooling",
+        "metadata": {
+            "type": "number",
+            "readable": True,
+            "writeable": True,
+            "unit": "°F",
+            "ccSpecific": {"setpointType": 2},
+        },
+        "value": 73,
+    }
+    assert args["value"] == 86
+
+    client.async_send_command.reset_mock()
+
+    with pytest.raises(ValueError):
```

Comment:
Can we test an invalid hvac mode for the hvac mode service too?

Reply:
added a test for an invalid preset mode too